### PR TITLE
Remove addLink that accepts Link interface

### PR DESCRIPTION
**Pull request by bogdandrutu**
Currently the Link interface cannot easily be removed, used in the Sampler interface,
will do a separate PR for that.

Based on the specs we don't have to offer a "lazy" formatted Link API (was removed some time ago).

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

**Review by jkwatson — APPROVED** [review]
very happy to make the API surface smaller, pre release.
Thanks!

File: api/src/main/java/io/opentelemetry/trace/Span.java [hunk 1]
```diff
@@ -444,10 +444,13 @@ default void setAttribute(AttributeKey<Long> key, int value) {
     /**
      * Adds a {@link Link} to the newly created {@code Span}.
      *
+     * <p>Links are used to link {@link Span}s in different traces. Used (for example) in batching
```
Comment (anuraaga) [inline api/src/main/java/io/opentelemetry/trace/Span.java:447]: Let's copy this to the other overload too.

**Review by anuraaga — APPROVED** [review]
Thanks!

File: api/src/main/java/io/opentelemetry/trace/Span.java (hunk 1 above)
Comment (bogdandrutu) [reply api/src/main/java/io/opentelemetry/trace/Span.java:447]: Done.

**Pull request merged**